services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/SegmentedInMemoryKeyValueStorage.java

```diff
@@ -111,6 +112,30 @@ public Optional<byte[]> get(final SegmentIdentifier segmentIdentifier, final byt
     }
   }
 
+  @Override
+  public Optional<NearestKeyValue> getNearestTo(
```

Comment:
I think we need to find another way to do that. it will be huge work for snapsync as a server

Reply:
Although we could use getNearest and return an iterator for snap sync server, I am thinking streamFlatAccounts and streamFlatStorage should work better and more easily for snapsync server.  

getNearest is really just for point-wise access to keys when in archive mode.

Reply:
ah yes got it not sure why I was thinking this one will be used for snapsync :)